### DISCUSSION TITLE
Connection refused (forge create SimpleStorage --rpc-url http://127.0.0.1:7545 --interactive)

### DISCUSSION THREAD

**Discussion by Souhademi** [Cyfrin/foundry-full-course-cu#873] [Q&A]
forge create SimpleStorage --rpc-url http://127.0.0.1:7545 --interactive
[⠊] Compiling...
No files changed, compilation skipped
Error: 
error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)

**Comment by Niferu** [comment]
Hi @Souhademi 

Have you ran `anvil` first of all to create local network?

If you did see below:

You are probably just listening to wrong url. Once you run `anvil` check below:

![image](https://github.com/foundry-rs/foundry/assets/110537615/2117b464-4448-4000-8c75-ffd35bdab619)

then in your command use proper url and it should work

**Reply by ChristianWupp** [reply]
Run, # forge script script/DeploySimpleStorage.s.sol --rpc-url http://127.0.0.1:8545

'''
Error: 
error sending request for url (http://127.0.0.1:8545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
'''

While being, "Listening on 127.0.0.1:8545" from anvil

Anvil Blockchain not running?

**Comment by Mindsetter** [comment]
Hi @Souhademi 

In other to help you resolve this issue I assume you have installed Ganache on Windows PC. If that is the case kindly follow these steps:

1. Open the Ganache Application and click on the settings icon (top right).
![1](https://github.com/Cyfrin/foundry-full-course-f23/assets/45667432/142a1b0d-e750-4ea8-9736-d6963bc8c003)

2. Click on server and change the hostname to ...vEthernet(WSL).
 
![2](https://github.com/Cyfrin/foundry-full-course-f23/assets/45667432/6c154e23-0dc6-463c-8b92-9247a3304f31)

3. This will change your RPC SERVER URL to  HTTP://172.27.112.1:7545 ( Kindly confirm from your Ganache Application).

4. Now use  http://172.27.112.1:7545 for your rpc-url.

Example 
``` 
forge create SimpleStorage --rpc-url http://172.27.112.1:7545 --interactive
```
NB: Don't close your Ganache Application.

Let me know if it works.
Thank you

**Reply by ShawnFromTexas** [reply]
this worked for me on exact issue frend! many thanks

**Comment by Nikzadkhan** [comment]
it works , thanks @Nlferu  so much
